### ISSUE TITLE
Implement getInternalsDomains methods for all frameworks

### ISSUE THREAD

**Issue by Naktibalda** [Codeception/Codeception#2396]
This is a follow up to https://github.com/Codeception/Codeception/pull/2371

I created a new branch: https://github.com/Codeception/Codeception/tree/feature-get-internal-domains
All frameworks that support multiple domain names in tests must implement [getInternalDomain](https://github.com/Codeception/Codeception/blob/feature-get-internal-domains/src/Codeception/Module/Yii1.php#L179) method. 

## ToDo

- [x] Laravel4
- [x] Laravel5
- [x] Lumen
- [x] Silex
- [x] Symfony2
- [x] Phalcon1
- [x] Phalcon2
- [x] Yii1
- [x] Yii2

**Comment by janhenkgerritsen** [comment]
Thanks for setting up the branch, I will try to implement this as soon as possible.

**Comment by Naktibalda** [comment]
A better place for default implementation of getInternalDomain method would be a [Codeception\Lib\Framework](https://github.com/Codeception/Codeception/blob/2.1/src/Codeception/Lib/Framework.php) class or a new common parent of all Connector classes.

And we could get rid of interface.

**Comment by Naktibalda** [comment]
I implemented getInternalDomains for ZF2,
code: https://github.com/Codeception/Codeception/pull/2405
tests: https://github.com/Naktibalda/codeception-zf2-tests/pull/2

An important observation is that domains are not limited to literal values, so in_array is not enough.
ZF2 supports placeholders in domain route and uses regex for matching.

My implementation of getInternalDomains returns array of patterns.
Can you do something similar for Laravel?

**Comment by janhenkgerritsen** [comment]
I implemented the `getInternalDomains()` method for the Laravel 5 module, and I also added tests for the functionality to the sample application: https://github.com/janhenkgerritsen/codeception-laravel5-sample/commit/068714847a21738414406ca0c2929d374eb45bb6.

To get this to work I had to use reflection. So I introduced a `Codeception\Util\ReflectionHelper` class. This class has two static methods. The first method is the `readPrivateProperty` method of the `PropertyAccess` class. The second method is a new method called `invokePrivateMethod`. I deprecated the `PropertyAccess::readPrivateProperty` method for now, and replaced the implementation, it now delegates to the method on the `ReflectionHelper` class.

If this feature branch is merged into the 2.1 branch we should replace all references to `PropertyAccess::readPrivateProperty` with `ReflectionHelper::readPrivateProperty`. But should we also remove the PropertyAccess class, or do we keep it for backwards compatability? I think we can remove it, because it hasn't been in the code very long and with normal Codeception usage you would not use this class. So mentioning this in the changelog should be enough, and the fix is very easy for people that do run into problems. What do you guys think?

**Comment by janhenkgerritsen** [comment]
Just implemented the method for the Laravel 4 module. I don't have the time to implement the method for the Lumen module today. Will do that asap.

**Comment by janhenkgerritsen** [comment]
Lumen does not support domain routing, so I deleted the `getInternalDomains()` method from the Lumen module.

**Comment by raistlin** [comment]
Implementation for Symfony2 https://github.com/Codeception/Codeception/pull/2420

**Comment by janhenkgerritsen** [comment]
Is someone familiar with Silex? I am not, but if nobody else is I can look into the implementation for the Silex module. Should not be too difficult I think.

**Comment by janhenkgerritsen** [comment]
@samdark Does Yii support domain routing? And if so, can you implement the functionality for this feature for the Yii modules?

I will look into the implementation for the Silex module.

**Comment by janhenkgerritsen** [comment]
I just implemented the `getInternalDomains` method for the Silex module.

**Comment by sergeyklay** [comment]
Implementation for Phalcon1 #2457 (Phalcon2 extends Phalcon1)

**Comment by Naktibalda** [comment]
Are we waiting for anything?

**Comment by janhenkgerritsen** [comment]
We are waiting on the implementation for Yii.

**Comment by Naktibalda** [comment]
@samdark are you aware that you are blocking this process?

**Comment by samdark** [comment]
Yes. Can't do anything about it atm. Busy with Yii 2.0.7 release.

**Comment by samdark** [comment]
Found some time. I have a problem implementing it. Both Yii 1.1 and Yii 2.0 support wildcards for domain names such as:

```
http://<name>.example.com/test
http://example.<name>.com/test
```

What should be returned in this case?

**Comment by samdark** [comment]
Code for Yii 1.1 and 2.0 is in the branch but parametrized routes are now simply ignored :(

**Comment by Naktibalda** [comment]
Return PCRE patterns, like here: https://github.com/Codeception/Codeception/blob/feature-get-internal-domains/src/Codeception/Module/Silex.php#L111

**Comment by samdark** [comment]
OK. Btw., does Silex module properly escape dots in regex generated?

**Comment by samdark** [comment]
Finished Yii1 and Yii2 modules. `getInternalDomains()` of these return ready to use properly escaped regexes such as:

```
/^http\://example\.com$/u
/^\w+\.example\.com$/u
/^[^\/]+\.example\.com$/u
```

**Comment by janhenkgerritsen** [comment]
I updated the regex for the Silex module by adding a call to `preg_quote`, good catch.

**Comment by samdark** [comment]
Checked travis errors. Seems these are releated with what already fixed in master, not with new code.

**Comment by Naktibalda** [comment]
This branch can't be merged to 2.1 automatically. Could somebody rebase it?

**Comment by janhenkgerritsen** [comment]
I merged this branch into 2.1 in #2611.
